pandas/io/common.py

```diff
@@ -115,25 +123,24 @@ def _stringify_path(filepath_or_buffer):
     Any other object is passed through unchanged, which includes bytes,
     strings, buffers, or anything else that's not even path-like.
     """
-    try:
-        import pathlib
-        _PATHLIB_INSTALLED = True
-    except ImportError:
-        _PATHLIB_INSTALLED = False
-
     try:
         from py.path import local as LocalPath
         _PY_PATH_INSTALLED = True
     except ImportError:
         _PY_PATH_INSTALLED = False
 
     if hasattr(filepath_or_buffer, '__fspath__'):
-        return filepath_or_buffer.__fspath__()
-    if _PATHLIB_INSTALLED and isinstance(filepath_or_buffer, pathlib.Path):
-        return str(filepath_or_buffer)
+        # mypy lacks comprehensive support for hasattr; see mypy#1424
+        # TODO (PY36): refactor to use os.PathLike
+        return filepath_or_buffer.__fspath__()  # type: ignore
     if _PY_PATH_INSTALLED and isinstance(filepath_or_buffer, LocalPath):
         return filepath_or_buffer.strpath
-    return _expand_user(filepath_or_buffer)
```

Comment:
I had to refactor this because Mypy wasn't narrowing types at this point and still guessed that `Path` objects could be returned here based off of the signature of `_expand_user` even though the preceding isinstance check wouldn't allow that. 

I commented on https://github.com/python/mypy/issues/6847#issuecomment-493712164 so will see if that gets any attention, but the refactor works as well